### PR TITLE
Add urdfdom_headers recipe.

### PR DESCRIPTION
**Pull request by lesteve**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

@jcarpent another comment needed about being a maintainer for this recipe.

### CODE REVIEW

**Comment by jcarpent** [comment]
I do agree to be in the maintainer list of this package.

File: recipes/urdfdom_headers/build.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+mkdir build
+cd build
+
+cmake .. \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_PREFIX_PATH=$PREFIX -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_INSTALL_LIBDIR=lib
```
Comment (marcelotrevisani) [inline recipes/urdfdom_headers/build.sh]: Should it be ``-DCMAKE_INSTALL_LIBDIR=$PREFIX/lib`` ?

File: recipes/urdfdom_headers/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - make
+  host:
+  run:
```
Comment (marcelotrevisani) [inline recipes/urdfdom_headers/meta.yaml]: please remove the unused sections

File: recipes/urdfdom_headers/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  home: http://github.com/ros/{{ name }}
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE
+  summary: 'Headers for URDF parsers'
+  dev_url: https://github.com/ros/{{ name }}
+
+extra:
+  recipe-maintainers:
+    - jcarpent
```
Comment (marcelotrevisani) [inline recipes/urdfdom_headers/meta.yaml]: @jcarpent , could you please confirm that you also want to be a maintainer of this package on ``conda-forge`` ?

Comment (lesteve) [reply recipes/urdfdom_headers/meta.yaml]: He agree in https://github.com/conda-forge/staged-recipes/pull/8911#issuecomment-514531321

File: recipes/urdfdom_headers/build.sh (hunk 1 above)
Comment (lesteve) [reply recipes/urdfdom_headers/build.sh]: I am not a cmake expert by any means, but I have seen [this](https://github.com/conda-forge/arrow-cpp-feedstock/issues/61) for example that seem to indicate that `lib` is preferred.

**Comment by wolfv** [comment]
I also agree.

File: recipes/urdfdom_headers/build.sh (hunk 1 above)
Comment (xhochy) [reply recipes/urdfdom_headers/build.sh]: No, `lib` is correct, `CMAKE_INSTALL_PREFIX` will be prepended by CMake.

Comment (lesteve) [reply recipes/urdfdom_headers/build.sh]: Great to have the confirmation, thanks!

File: recipes/urdfdom_headers/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    - {{ compiler('cxx') }}
+    - cmake
+    - make
+
+test:
+  requires:
+    - pkg-config
+  commands:
+    - pkg-config --modversion {{ name }}
+    - ls $PREFIX/include/urdf_*/*.h
```
Comment (lesteve) [inline recipes/urdfdom_headers/meta.yaml]: If there is no `.h` files I am expecting this `ls` command to exit with a non-zero exit code and the test to fail. I checked locally that this was the case with `conda-build`.

A suggestion of a better test to use for header-only packages would be more than welcome.

Comment (xhochy) [reply recipes/urdfdom_headers/meta.yaml]: You could explicitly check for a named header.

Comment (lesteve) [reply recipes/urdfdom_headers/meta.yaml]: OK I have done that in my last commit.

**Pull request merged**